news-frontend/src/contexts/ArticleContext.js

```diff
@@ -0,0 +1,140 @@
+import newsApi from "../utils/NewsApi";
+import mainApi from "../utils/MainApi";
+import { useLoggedIn } from "./LoggedInContext";
+
+const { createContext, useContext, useState, useEffect } = require("react");
+const ArticlesContext = createContext();
+
+const ArticleContextProvider = ({ children }) => {
+  const [keyword, setKeyword] = useState("");
+  const [notFound, setNotFound] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [cards, setCards] = useState([]);
+  const [toShowMore, setToShowMore] = useState(3);
+  const [isSaveCards, setIsSaveCards] = useState([]);
+
+  const { token, isLoggedIn } = useLoggedIn();
+
+  //////////////////////////////////INITIALL-USE-EFFECTS/////////////////////
+
+  useEffect(() => {
+    const token = localStorage.getItem("jwt");
+    if (!isLoggedIn) {
+      setIsSaveCards([]);
+      return;
+    } 
+    mainApi.getSavedArticles(token)
+    .then((res) => {
+      setIsSaveCards(res);
+    });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**

Reply:
thank for the comment, issue fixed.